### ISSUE TITLE
[Controller Request]: 

### ISSUE THREAD

**Issue by 664710775** [ndeadly/MissionControl#882]
### Controller Name

黑鲨双翼手柄

### Current Controller Behaviour

Controller is not successful in pairing with the console

### Supported Features

- [ ] Rumble/Vibration
- [ ] Motion Controls
- [ ] Player Number Indicator
- [ ] Battery Level Reporting

**Comment by ndeadly** [comment]
English only, please.

**Comment by 664710775** [comment]
I am unable to determine the English name of the controller. You can call it the Black Shark biplane handle. This is a paired joystick similar to Joycon.
![7c1bd7dd4b1a07e9d907a4157199c7b1](https://github.com/user-attachments/assets/e2a43a13-8e19-4a70-8ed4-eb74f96a2915)

**Comment by ndeadly** [comment]
That's ok, I just want to have some idea of what this controller is. Do you have a link to a user manual or something? (Chinese is fine, I can translate it)

Does the controller connect to the system at all? Many newer controllers are using Bluetooth LE, which isn't supported in the release versions of Mission Control. I have builds supporting BLE available on my discord server that you could try to connect it with.

**Comment by 664710775** [comment]
![01](https://github.com/user-attachments/assets/70f6abf3-23d8-4629-afb8-2c5afc4a0cb4)
![02](https://github.com/user-attachments/assets/af271b1b-1f77-4ee4-92e1-3b231a477498)
![03](https://github.com/user-attachments/assets/c1beb416-53d7-4b7b-9fe5-b5ee2f535c02)
https://pan.baidu.com/s/1SNt9tMUUFtm4Z38A-b7jHw
These are screenshots of some manuals and cracked firmware for the left controller. The default mapping for the directional keys on the left joystick is ABXY, and it is estimated that additional mapping changes will be required. But I don't have enough professional knowledge to access and write these Bin files. I will try to see if the BLE version can be linked.

**Comment by ndeadly** [comment]
Haha, images might be a little harder to run through a translator :p. But I see 4.2 and 4.4 listed in there, which are likely what I was looking to see (bluetooth version numbers). 4.0+ usually indicates that it's a BLE controller, so yeah, give the discord build a try (see #testing-ble channel). Note that there is currently something wrong with the patches on firmware 19.0.x, so you will need to use it on a firmware < 19.0.0 for now until I have time to fix it

**Comment by 664710775** [comment]
17.0.1 AMS1.6.2 Which version am I using? I tried 0.10.0 but still got an error.

**Comment by ndeadly** [comment]
Yes, the 0.10.0 version should be the one you want for fw 17.x.x. Though i couldn't really tell you what stability improvements were made since then. You would be best running an 0.11.0+ version on 18.x.x. What's the error you're getting?

**Comment by 664710775** [comment]
I used 0.11.0 in 18.1.0 happend this error
![IMG_20241031_210608](https://github.com/user-attachments/assets/b3482871-a7e5-4ac4-8cd3-0a4022ddfa6d)

**Comment by ndeadly** [comment]
That can happen from time to time, it's not entirely stable. If it happened while pairing the controller, there is possibly useful in formation in the log (sdmc:/mc-mitm.log). If it happened randomly you might just need to try again.

**Comment by 664710775** [comment]
This appears when opening the atmospheric virtual system

**Comment by ndeadly** [comment]
Try clearing the pairing database, and make sure you're not connecting any other wireless controllers as they might be increasing the likelihood of the error

**Comment by 664710775** [comment]
It worked,thanks. But still unable to search for the controller.It may be an issue with the controller itself

**Comment by ndeadly** [comment]
Share the log it created, it will show what's happening.

**Comment by 664710775** [comment]
Perhaps because this controller alone cannot RF out the L+R button signal, pressing L+R separately from left and right will not be considered as a paired Joycon.

**Comment by ndeadly** [comment]
Not sure what you mean. Did you get it connected?

**Comment by 664710775** [comment]
![IMG_20241031_215020](https://github.com/user-attachments/assets/008012fb-a5db-450a-9776-bdc62756831e)
Unable to search for these two controllers

**Comment by 664710775** [comment]
It seems like something has been detected, but the switch is now stuck on this interface....

**Comment by 664710775** [comment]
I forced it to shut down directly. Where can I find the logs

**Comment by ndeadly** [comment]
I told you here https://github.com/ndeadly/MissionControl/issues/882#issuecomment-2449815627

sdmc:/mc-mitm.log

**Comment by 664710775** [comment]
[mc-mitm[2].log](https://github.com/user-attachments/files/17588766/mc-mitm.2.log)

**Comment by ndeadly** [comment]
I'm not seeing any BLE events in there, just a bunch of regular inquiry packets from what looks to be a phone. Are you sure you're entering pairing mode correctly on the controller, and that it does indeed support bluetooth?

**Comment by 664710775** [comment]
The device BTHLE \ Dev_04338505fc0a \ 8&DB2a817&0&04338505fc0a has been configured.
Driver program name: bthleenum. info
Driver package ID: bthleenum. inf_amd64_33b8a785b5a06c38
Class UID: {e0cbf06c-CD8b-4647-bb8a-263b43f0f974}
Driver Program Date: 06/21/2006
Driver version: 10.026100.2130
Driver provider: Microsoft
Driver section: BthLEEnum Device.NT
Driver level: 0xFF2000
Matching Device ID: BTHLE \ GenericDevice
Overlimit driver program:
Device updated: false
Parent device: BTH \ MS-BTHLE \ 7&12f646e8&4&3

**Comment by ndeadly** [comment]
That certainly looks like an LE device from that description. This is paired to your PC?

**Comment by 664710775** [comment]
Yes, and the computer can only connect to one controller at a time

**Comment by ndeadly** [comment]
Bizzarre. How are you supposed to use these controllers in practice then? I still don't really understand why they're not showing up in the log either. If they're BLE there should at least be _some_ activity, even if the connection is ultimately unsuccessful

**Comment by 664710775** [comment]
Android phones can be linked normally and mapped for use through some software. The mobile phones under the controller production company also have specialized adaptation software
